pkgs/os-specific/darwin/libtapi/default.nix

```diff
@@ -9,22 +9,36 @@ stdenv.mkDerivation {
     sha256 = "1a19h39a48agvnmal99n9j1fjadiqwib7hfzmn342wmgh9z3vk0g";
   };
 
+  sourceRoot = "source/src/llvm";
+
   nativeBuildInputs = [ cmake python3 ];
-  buildInputs = [ clang_6.cc ];
 
+  # ncurses is required here to avoid a reference to bootstrap-tools, which is
+  # not allowed for the stdenv.
+  buildInputs = [ ncurses ];
+
+  cmakeFlags = [ "-DLLVM_INCLUDE_TESTS=OFF" ];
+
+  # fixes: fatal error: 'clang/Basic/Diagnostic.h' file not found
+  # adapted from upstream
+  # https://github.com/tpoechtrager/apple-libtapi/blob/3cb307764cc5f1856c8a23bbdf3eb49dfc6bea48/build.sh#L58-L60
   preConfigure = ''
-    cd src/llvm
+    INCLUDE_FIX="-I $PWD/projects/clang/include"
+    INCLUDE_FIX+=" -I $PWD/build/projects/clang/include"
+
+    cmakeFlagsArray+=(-DCMAKE_CXX_FLAGS="$INCLUDE_FIX")
   '';
 
-  cmakeFlags = [ "-DLLVM_INCLUDE_TESTS=OFF" ];
+  buildFlags = [ "clangBasic" "libtapi" ];
```

Comment:
do you need clangBasic here? we're already building it elsewhere, so i was hoping we could leave it off

Reply:
This is required after removing `clang_6.cc` from buildInputs. I'm guessing it's an undeclared dependency in the build system. If this isn't specified then the build fails with:

```
  In file included from /tmp/nix-build-libtapi-1000.10.8.drv-0/source/src/llvm/projects/libtapi/lib/Core/Architecture.cpp:18:
  In file included from /private/tmp/nix-build-libtapi-1000.10.8.drv-0/source/src/llvm/projects/clang/include/clang/Basic/Diagnostic.h:18:
  /private/tmp/nix-build-libtapi-1000.10.8.drv-0/source/src/llvm/projects/clang/include/clang/Basic/DiagnosticIDs.h:71:10: fatal error: 'clang/Basic/DiagnosticCommonKinds.inc' file not found
  #include "clang/Basic/DiagnosticCommonKinds.inc"
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
```

> we're already building it elsewhere, so i was hoping we could leave it off

I'm making a bit of an assumption that "clangBasic" is a small internal clang target and not something that should be shared between builds, but a quick web search doesn't show up documentation for what it contains. The [library definition][clangBasic] is fairly small.

[clangBasic]: https://github.com/llvm/llvm-project/blob/9b04fec0021112d913058e81bf864e4f33c33dcc/clang/lib/Basic/CMakeLists.txt